workers/unity/Assets/Playground/Scripts/UI/InitUISystem.cs

```diff
@@ -45,5 +45,10 @@ protected override void OnUpdate()
             // Disable system after first run.
             Enabled = false;
         }
+
+        protected override void OnDestroyManager()
+        {
+            UnityEngine.Object.Destroy(UIComponent.Main.gameObject);
```

Comment:
use the `UnityObjectDestroyer`

Reply:
Interesting, didn't know about that one. Not related, but I am curious whey do we use DestoyImmediate in tests